v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/SpannerToSourceDb.java

```diff
@@ -526,19 +527,22 @@ public static PipelineResult run(Options options) {
     shadowTableCreator.createShadowTablesInSpanner();
     Ddl ddl = SpannerSchema.getInformationSchemaAsDdl(spannerConfig);
     List<Shard> shards;
+    ShardConfig shardConfig;
     String shardingMode;
     if (MYSQL_SOURCE_TYPE.equals(options.getSourceType())) {
       ShardFileReader shardFileReader = new ShardFileReader(new SecretManagerAccessorImpl());
-      shards = shardFileReader.getOrderedShardDetails(options.getSourceShardsFilePath());
+      shardConfig = shardFileReader.getOrderedShardDetails(options.getSourceShardsFilePath());
+      shards = shardConfig.getShards();
       shardingMode = Constants.SHARDING_MODE_MULTI_SHARD;
 
     } else {
       CassandraConfigFileReader cassandraConfigFileReader = new CassandraConfigFileReader();
       shards = cassandraConfigFileReader.getCassandraShard(options.getSourceShardsFilePath());
+      shardConfig = new ShardConfig(shards, false);
       LOG.info("Cassandra config is: {}", shards.get(0));
       shardingMode = Constants.SHARDING_MODE_SINGLE_SHARD;
     }
-    if (shards.size() == 1) {
+    if (shards.size() == 1 && !shardConfig.isShardedMigration()) {
```

Comment:
I think this needs to be a template level flag and not a shard level configuration..